### PR TITLE
Adding deprecation warning for the Http package

### PR DESCRIPTION
**Pull request by burzum**
Refs #11385

### CODE REVIEW

**Comment by markstory** [comment]
I will work on getting the tests updated over the next few days as there are quite a few uses.

**Comment by burzum** [comment]
@markstory please take a look at this commit https://github.com/cakephp/cakephp/pull/11414/commits/3f1d4dac10358f6b87edd9038b1789da309e8554 I had a short discussion with @lorenzo about this. If you want to do it different feel free to change it.

File: src/Http/ServerRequest.php [hunk 1]
```diff
@@ -625,6 +635,11 @@ public function __call($name, $params)
      */
     public function __get($name)
     {
+        deprecationWarning(
+            'Accessing routing parameters through __get will removed in 4.0.0. ' .
```
Comment (ADmad) [inline src/Http/ServerRequest.php]: A message like `"Accessing '{$name}' as property has been deprecated and will be removed in 4.0.0"` might be more easily understandable to novice users.

**Comment by markstory** [comment]
> please take a look at this commit 3f1d4da I had a short discussion with @lorenzo about this. If you want to do it different feel free to change it.

I think what you have is reasonable for that given the constraints we're facing.

File: tests/TestCase/Controller/Component/RequestHandlerComponentTest.php [hunk 2]
```diff
@@ -542,37 +553,44 @@ public function testStartupProcessData()
...
     /**
      * Test mapping a new type and having startup process it.
      *
+     * @group deprecated
      * @return void
      * @triggers Controller.startup $this->Controller
      */
     public function testStartupCustomTypeProcess()
     {
-        $restore = error_reporting(E_ALL & ~E_USER_DEPRECATED);
```
Comment (saeideng) [inline tests/TestCase/Controller/Component/RequestHandlerComponentTest.php]: please re-base 
I removed this line in other PR
`testStartupCustomTypeProcess` should not flag `@group deprecated`

Comment (markstory) [reply tests/TestCase/Controller/Component/RequestHandlerComponentTest.php]: Thanks!

File: tests/test_app/TestApp/Controller/TestsAppsController.php [hunk 3]
```diff
@@ -62,9 +60,7 @@ public function redirect_to_permanent()
 
     public function set_type()
     {
-        $this->response->type('json');
-
-        return $this->response;
+        return $this->response->withType('json');
```
Comment (saeideng) [inline tests/test_app/TestApp/Controller/TestsAppsController.php]: is this true?
```php
       $this->response->type('json');

        return $this->response;
```
will change origin `response` and returns (mutable )
```php
$this->response->withType('json');
```
will no change origin `response` and returns (immutable )


I mean

```php
// value of $this->response
$responseResult=set_type()

$this->response == $responseResult //before
$this->response != $responseResult //now
```

I know this is just a testCase

**Comment by saeideng** [comment]
` Files changed 66 ` until now :clap:

File: tests/test_app/TestApp/Controller/TestsAppsController.php (hunk 3 above)
Comment (markstory) [reply tests/test_app/TestApp/Controller/TestsAppsController.php]: You're right this doesn't modify the controller's `response` property. In this situation that isn't relevant as a controller action returning a response overrides the property.

**Comment by markstory** [comment]
@saeideng Almost below 100 failing tests as well.

**Comment by ADmad** [comment]
:white_check_mark:  :tada:

**Pull request merged**